app/controllers/stores_controller.rb

```diff
@@ -1,5 +1,9 @@
 class StoresController < ApplicationController
   def show
-    render layout: 'wide',  :locals => {:background => "armory"}
+    location = Location.find_by(slug: params[:location_slug])
+    store_type = params[:slug]
+    @store = location.send(store_type)
+
+    render layout: 'wide',  :locals => {:background => store_type}
```

Comment:
Unnecessary spacing detected.<br>Use the new Ruby 1.9 hash syntax.<br>Space inside { missing.<br>Space inside } missing.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.